### PR TITLE
C++11 cleanup on constructors dir client

### PR DESCRIPTION
**Pull request by Dumbeldor**

### CODE REVIEW

File: src/client/inputhandler.h [hunk 1]
```diff
@@ -277,24 +276,15 @@ class RealInputHandler : public InputHandler
...
-	{
-		leftdown = false;
-		rightdown = false;
-		leftclicked = false;
-		rightclicked = false;
-		leftreleased = false;
-		rightreleased = false;
-		keydown.clear();
-	}
+	RandomInputHandler(){}
```
Comment (nerzhul) [inline src/client/inputhandler.h]: missing space

File: src/client/inputhandler.h [hunk 2]
```diff
@@ -277,24 +276,15 @@ class RealInputHandler : public InputHandler
 	}
 
 private:
-	IrrlichtDevice *m_device;
-	MyEventReceiver *m_receiver;
-	v2s32 m_mousepos;
+	IrrlichtDevice *m_device = nullptr;
+	MyEventReceiver *m_receiver = nullptr;
+	v2s32 m_mousepos = v2s32();
```
Comment (nerzhul) [inline src/client/inputhandler.h]: useless constructor

File: src/client/inputhandler.h [hunk 3]
```diff
@@ -141,34 +141,33 @@ class MyEventReceiver : public IEventReceiver
...
+	JoystickController *joystick = nullptr;
 
 #ifdef HAVE_TOUCHSCREENGUI
 	TouchScreenGUI *m_touchscreengui;
 #endif
 
 private:
 	// The current state of keys
-	KeyList keyIsDown;
+	KeyList keyIsDown = {};
```
Comment (nerzhul) [inline src/client/inputhandler.h]: both {} are useless

File: src/client/joystick_controller.cpp [hunk 4]
```diff
@@ -154,12 +154,8 @@ JoystickLayout create_xbox_layout()
 	return jlo;
 }
 
-JoystickController::JoystickController()
+JoystickController::JoystickController() : doubling_dtime(g_settings->getFloat("repeat_joystick_button_time"))
```
Comment (nerzhul) [inline src/client/joystick_controller.cpp]: line too long

File: src/client/tile.h [hunk 5]
```diff
@@ -180,43 +180,19 @@ enum MaterialType{
...
-		texture(NULL),
-		normal_texture(NULL),
-		flags_texture(NULL)
-	{
-	}
-	u32 texture_id;
-	video::ITexture *texture;
-	video::ITexture *normal_texture;
-	video::ITexture *flags_texture;
+	FrameSpec()	{}
```
Comment (nerzhul) [inline src/client/tile.h]: too many spaces

File: src/client/tile.h [hunk 6]
```diff
@@ -180,43 +180,19 @@ enum MaterialType{
...
-			//0 // <- DEBUG, Use the one below
-			MATERIAL_FLAG_BACKFACE_CULLING |
-			MATERIAL_FLAG_TILEABLE_HORIZONTAL|
-			MATERIAL_FLAG_TILEABLE_VERTICAL
-		),
-		has_color(false),
-		color()
-	{
-	}
+	TileLayer()	{}
```
Comment (nerzhul) [inline src/client/tile.h]: same here

File: src/client/tile.h [hunk 7]
```diff
@@ -287,24 +263,28 @@ struct TileLayer
...
+		MATERIAL_FLAG_BACKFACE_CULLING |
+		MATERIAL_FLAG_TILEABLE_HORIZONTAL|
+		MATERIAL_FLAG_TILEABLE_VERTICAL);
 
 	//! If true, the tile has its own color.
-	bool has_color;
+	bool has_color = false;
 
-	std::vector<FrameSpec> frames;
+	std::vector<FrameSpec> frames = {};
```
Comment (nerzhul) [inline src/client/tile.h]: useless {}

File: src/client/inputhandler.h [hunk 8]
```diff
@@ -387,15 +377,15 @@ class RandomInputHandler : public InputHandler
 	s32 Rand(s32 min, s32 max);
 
 private:
-	KeyList keydown;
+	KeyList keydown = {};
```
Comment (nerzhul) [inline src/client/inputhandler.h]: useless

File: src/client/tile.h [hunk 9]
```diff
@@ -287,22 +263,26 @@ struct TileLayer
...
 
-	u16 animation_frame_length_ms;
-	u8 animation_frame_count;
+	u16 animation_frame_length_ms = 0;
+	u8 animation_frame_count = 1;
 
-	u8 material_type;
-	u8 material_flags;
+	u8 material_type = TILE_MATERIAL_BASIC;
+	u8 material_flags = (
```
Comment (nerzhul) [inline src/client/tile.h]: useless parenthesis

**Comment by Zeno-** [comment]
:+1:

**Pull request merged**